micro-rdk/src/common/entry.rs

```diff
@@ -228,7 +232,7 @@ pub async fn serve_inner<S>(
         .unwrap();
 
     // Attempt to cache the config for the machine we are about to `serve`.
-    if let Err(e) = storage.store_robot_configuration(cfg_response.config.unwrap()) {
+    if let Err(e) = storage.store_robot_configuration(cfg_response.config.unwrap_or_default()) {
```

Comment:
in the absence of config we should never cache an empty config as it would be an invalid state